### PR TITLE
Allow the use of a custom base OID for the subobject of each trap.

### PR DESCRIPTION
**Pull request by vigno88**
Currently the base OID of each subobject is the content of the "oid" label of an alert with a ".2" appended and ".1", ".2", ".3" appended for each of the subobject. 

This PR adds a new cli parameter that would overwrite the base OID of each subobject with the parameter value. The ".1", ".2" and ".3" would still be appended to each sub-object as each one requires a different OID.

This allows to have multiple traps with different OIDs, which always having the sub-objects with constant OIDs.

### CODE REVIEW

**Comment by maxwo** [comment]
Hi,

Thanks for your interest. Can you describe more precisely the use-case of such an option? And how `snmp.custom-base-oid-sub-object` or the alert labels cannot be used?

Thanks.

**Comment by vigno88** [comment]
Hi!

Let's say each of our trap has a different OID and the trap sub-objects are type of objects described in the MIBs (meaning they have a constant OID). Then the label cannot be used as a prefix for the sub-objects, because each label is a different OID, thus the OID of each sub-objects is always different depending of the trap type.

By allowing to define the trap sub object base oid, the trap OID can be different for each trap and the trap sub-objects can remain constant.

Nathan AV

File: README.md [hunk 1]
```diff
@@ -112,6 +112,8 @@ Flags:
                                  The alert severity if none is provided via labels.
       --snmp.version=V2c         SNMP version. V2c and V3 are currently supported.
       --snmp.timeout=5s          SNMP timeout
+      --snmp.custom-base-oid-sub-object=BASE_OID_SUB_OBJECT
```
Comment (maxwo) [inline README.md:115]: Can you rename to `--snmp.sub-object-oid`, and specify it defaults to `--snmp.trap-default-oid` or `--snmp.trap-oid-label`?

File: configuration/configuration.go [hunk 2]
```diff
@@ -52,6 +52,7 @@ func ParseConfiguration(args []string) (*SNMPNotifierConfiguration, log.Logger,
 		snmpTrapDescriptionTemplate = application.Flag("snmp.trap-description-template", "SNMP description template.").Default("description-template.tpl").ExistingFile()
 		snmpExtraFieldTemplate      = application.Flag("snmp.extra-field-template", "SNMP extra field templates, eg. --snmp.extra-field-templates=4=new-field.template.tpl to add a 4th field to the trap, with the given template file. You may add several fields using that flag several times.").PlaceHolder("4=extra-field-template.tpl").StringMap()
 		snmpTimeout                 = application.Flag("snmp.timeout", "SNMP timeout duration").Default("5s").Duration()
+		snmpCustomBaseOidSubObject  = application.Flag("snmp.custom-base-oid-sub-object", "OID to use as the base of the sub-objects of each trap.").PlaceHolder("BASE_OID_SUB_OBJECT").String()
```
Comment (maxwo) [inline configuration/configuration.go:55]: Use a "valid" OID as PlaceHolder, such as `1.3.6.1.4.1.123.456`

**Review by maxwo — CHANGES_REQUESTED** [review]
It seems like an interesting feature. I was just wondering how to provide a flexible way to customize trap OID and sub-objects OID independently, like providing an optional `--snmp.sub-object-oid-label="oid"` to retrieve the sub-object OID from the alerts themselves.

File: README.md (hunk 1 above)
Comment (vigno88) [reply README.md]: Fixed

File: configuration/configuration.go (hunk 2 above)
Comment (vigno88) [reply configuration/configuration.go]: Fixed

**Review by maxwo — APPROVED** [review]

**Comment by maxwo** [comment]
Merci pour la contribution

**Pull request merged**